python/lsst/pipe/tasks/quickFrameMeasurement.py

```diff
@@ -0,0 +1,405 @@
+        self.apFluxControl = measBase.ApertureFluxControl()
+        md = dafBase.PropertySet()
+        self.apFluxer = measBase.CircularApertureFluxAlgorithm(self.apFluxControl, "aperFlux",
+                                                               self.schema, md)
+
+        self.table = afwTable.SourceTable.make(self.schema)  # make sure to call this last!
+
+    @staticmethod
+    def detectObjectsInExp(exp, nSigma, nPixMin, grow=0):
+        """Run a very basic but fast threshold-based object detection on an exposure
+        Return the footPrintSet for the objects in a postISR exposure."""
+        threshold = afwDetect.Threshold(nSigma, afwDetect.Threshold.STDEV)
+        footPrintSet = afwDetect.FootprintSet(exp.getMaskedImage(), threshold, "DETECTED", nPixMin)
+        if grow > 0:
+            isotropic = True
+            footPrintSet = afwDetect.FootprintSet(footPrintSet, grow, isotropic)
+        return footPrintSet
+
+    @staticmethod
+    def checkResult(exp, centroid, srcNum, percentile):
+        """Perform a sanity check that centroid location is actually bright."""
+        threshold = np.percentile(exp.image.array, percentile)
+        pixelValue = exp.image[centroid]
+        if pixelValue < threshold:
+            msg = (f"Sanity check failed: srcNum {srcNum} at {centroid}"
+                   f" has central pixel = {pixelValue:3f} <"
+                   f" {percentile} percentile of image = {threshold:3f}")
+            raise ValueError(msg)
+        return
+
+    @staticmethod
+    def _calcMedianXxYy(objData):
+        """Return the median ixx and iyy for object in the image"""
+        medianXx = np.nanmedian([objData[i]['xx'] for i in objData.keys()])
+        medianYy = np.nanmedian([objData[i]['yy'] for i in objData.keys()])
+        return medianXx, medianYy
+
+    def _calcBrightestObjSrcNum(self, objData):
+        """Find the brightest source which passes the cuts among the sources."""
+        max70, max70srcNum = -1, -1
+        max25, max25srcNum = -1, -1
+
+        for srcNum in sorted(objData.keys()):  # srcNum not contiguous so don't use a list comp
+            # skip flag used rather than continue statements so we have all the
+            # metrics computed for debug purposes as this task is whack-a-mole
+            skip = False
+            xx = objData[srcNum]['xx']
+            yy = objData[srcNum]['yy']
+
+            xx = max(xx, 1e-9)  # need to protect against division by zero
+            yy = max(yy, 1e-9)  # because we don't `continue` on zero moments
+
+            if self.config.doExtendednessCut:
+                if xx > self.config.maxExtendedness or yy > self.config.maxExtendedness:
+                    skip = True
+
+            nonRoundness = xx/yy
+            nonRoundness = max(nonRoundness, 1/nonRoundness)
+            if nonRoundness > self.config.maxNonRoundness:
+                skip = True
+
+            if self.log.isDebugEnabled():
```

Comment:
I didn't know either until I was prepping for this review, and found myself wanting a way to get out of being told not to use f-strings when it might not fire, and realised I could do this 🙂